File: 75-Blind/Java/MaxSubArray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if (nums.length == 0) { return -1; }
        
        int result = nums[0];
        int currSum = nums[0];
        
        for (int i = 1; i < nums.length; i++) {
            currSum = Math.max(nums[i], currSum + nums[i]);
            result = Math.max(currSum, result);
        }
        
        return result;
    }
}

File: 75-Blind/Python/MaxSubArray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) is 0:
            return 0
        currSum = nums[0]
        maxSum = nums[0]
        
        for i in range (1, len(nums)):
            currSum = max(nums[i], currSum + nums[i])
            maxSum = max(maxSum, currSum)
        
        return maxSum
